### DISCUSSION TITLE
Wix Custom action

### DISCUSSION THREAD

**Discussion by abhishekFisDevOps** [wixtoolset/issues#8773] [Questions]
### WiX Version

5.0.1

### .NET or MSBuild or Visual Studio Version

.NET 4

### HeatWave Version

1.0.4.5

### Windows Version

windows 2019 server

### Repro Steps

1. Create C# customaction to return INSTALL_CONDITION =13 OR 0 Based on condition
2. Created wxs file contain two component of files ex custom_component.txt ,mail_componet.txt 
3. apply condition on component custom_component -Id="MainAppl"    Guid="guid"   Condition="INSTALL_CONDITION =13" 
4 call customaction before installfiles -
InstallExecuteSequence
  Custom Action ="SetInstallCondition" Before="InstallFiles" //alo try some event like -CostInitialize
  Custom>

5-in Logs property is setting correct value

### Actual Result

custom_component not installing based on component 

### Expected Result

if custom action returning 13 then it should copy files under custom_component
when same condition apply on InstallFiles event it works but it will not install any component , compoent level condition should also works
--condition on installfiles as below 
InstallExecuteSequence
  InstallFiles Condition="INSTALL_CONDITION=13" // It works ,and skipped installfile action if value is not 13 

### Acknowledgements

- [X] I acknowledge that this is a fully completed bug report. It is not a question or attempt to get help debugging my issue (because those should be sent to Discussions).

**Comment by barnson** [comment]
WiX doesn't use issues for support. We moved this issue to the [GitHub discussions area][gh]. You might also consider asking your question on [Stack Overflow][so].

[gh]: https://github.com/wixtoolset/issues/discussions?discussions_q=category%3AQuestions
[so]: https://stackoverflow.com/questions/tagged/wix

**Comment by bevanweiss** [comment]
If you're trying to alter variables used in a Component Condition then you'd need to do this before the CostFinalize.
or follow the guidance from here https://learn.microsoft.com/en-au/windows/win32/msi/component-table

> Condition
> 
> This column contains a conditional statement that can control whether a component is installed. If the condition is null or evaluates to true, then the component is enabled. If the condition evaluates to False, then the component is disabled and is not installed.
> 
> The Condition field enables or disables a component only during the [CostFinalize action](https://learn.microsoft.com/en-au/windows/win32/msi/costfinalize-action). To enable or disable a component after CostFinalize, you must use a custom action or the [DoAction ControlEvent](https://learn.microsoft.com/en-au/windows/win32/msi/doaction-controlevent) to call [MsiSetComponentState](https://learn.microsoft.com/en-us/windows/desktop/api/Msiquery/nf-msiquery-msisetcomponentstatea).
> 
> Note that unless the Transitive bit in the Attributes column is set for a component, the component remains enabled once installed even if the conditional statement in the Condition column later evaluates to False on a subsequent maintenance installation of the product.
> 
> The Condition column in the Component table accepts conditional expressions containing references to the installed states of features and components. For information on the syntax of conditional statements, see [Conditional Statement Syntax](https://learn.microsoft.com/en-au/windows/win32/msi/conditional-statement-syntax).
